### ISSUE TITLE
BingX - Edit order (for Stop loss) on an open position on BingX

### ISSUE THREAD

**Issue by xerno0om** [ccxt/ccxt#20840]
### Operating System

Windows 11

### Programming Languages

Python

### CCXT Version

Last

### Description

hello, can someone give me the code to edit a order (for the stop loss) on an position that is already open on bingX, Thank you,

For the moment I use this code, but the problem is that I have to add (or remove if I put 'sell') the quantity to set my stop loss, if you have a code which modifies an order or position, which is already open to add a stop loss I am interested

### Code

```
order_id = yours
symbol = "BTC-USDT"
amount = 42540.4
pourcentag = 0.38 / 100
quantity = 0.0001
stop_loss = amount - (amount * pourcentag)
stop_loss_formatted = round(stop_loss, 2)

params = {
    'position_id': order_id,
    "stopLoss": "{\"type\": \"STOP_MARKET\", \"quantity\": " + str(quantity) + ",\"stopPrice\": " + str(stop_loss_formatted) + ",\"price\": " + str(amount) + ",\"workingType\":\"MARK_PRICE\"}"
}
bybit_trade = bybit_exchange.create_market_order(symbol, 'buy', quantity, 90000, params=params)

print(bybit_trade)

```

**Comment by Dan-krm** [comment]
Hey @xerno0om if you want to add a stop loss or take profit to an already opened position on `bingx` you can use the `stopLossPrice` or `takeProfitPrice` parameters:

### BingX add stop loss or take profit to an open position:

```python
bingx_exchange = ccxt.bingx({
    'apiKey': 'YOUR_API_KEY',
    'secret': 'YOUR_SECRET',
    # 'verbose': True,  # for debug output
})

# change the values here
symbol = 'BTC/USDT:USDT'
type = 'market'
side = 'sell'
amount = 0.0001
price = None
stop_loss = 35000
take_profit = 50000

try:
    bingx_trade = await bingx_exchange.create_order(symbol, type, side, amount, price, {
        'stopLossPrice': 'stop_loss',
        'takeProfitPrice': 'take_profit',
    })
    pprint(bingx_trade)
except Exception as e:
    print('create_order() failed')
    print(e)

await exchange.close()
```

You can read more about placing orders on CCXT here: https://docs.ccxt.com/#/README?id=placing-orders

Unfortunately `editOrder` is not supported on bingx, but you might be able to use the `cancelOrder` method `bingx_exchange.cancel_order(id, symbol)` then create a new order using the above approach if you wanted to edit your stop loss or take profit values.

Trailing stop orders are also supported on bingx:
```python
bingx_trailing_amount_order = await bingx_exchange.create_order(symbol, type, side, amount, price, {
    'trailingAmount': 100,
    'reduceOnly': True,
})
```
```python
bingx_trailing_percent_order = await bingx_exchange.create_order(symbol, type, side, amount, price, {
    'trailingPercent': 10,
    'reduceOnly': True,
})
```

**Comment by xerno0om** [comment]
> Hey @xerno0om if you want to add a stop loss or take profit to an already opened position on `bingx` you can use the `stopLossPrice` or `takeProfitPrice` parameters:
> 
> ### BingX add stop loss or take profit to an open position:
> ```python
> bingx_exchange = ccxt.bingx({
>     'apiKey': 'YOUR_API_KEY',
>     'secret': 'YOUR_SECRET',
>     # 'verbose': True,  # for debug output
> })
> 
> # change the values here
> symbol = 'BTC/USDT:USDT'
> type = 'market'
> side = 'sell'
> amount = 0.0001
> price = None
> stop_loss = 35000
> take_profit = 50000
> 
> try:
>     bingx_trade = await bingx_exchange.create_order(symbol, type, side, amount, price, {
>         'stopLossPrice': 'stop_loss',
>         'takeProfitPrice': 'take_profit',
>     })
>     pprint(bingx_trade)
> except Exception as e:
>     print('create_order() failed')
>     print(e)
> 
> await exchange.close()
> ```
> 
> You can read more about placing orders on CCXT here: https://docs.ccxt.com/#/README?id=placing-orders
> 
> Unfortunately `editOrder` is not supported on bingx, but you might be able to use the `cancelOrder` method `bingx_exchange.cancel_order(id, symbol)` then create a new order using the above approach if you wanted to edit your stop loss or take profit values.
> 
> Trailing stop orders are also supported on bingx:
> 
> ```python
> bingx_trailing_amount_order = await bingx_exchange.create_order(symbol, type, side, amount, price, {
>     'trailingAmount': 100,
>     'reduceOnly': True,
> })
> ```
> 
> ```python
> bingx_trailing_percent_order = await bingx_exchange.create_order(symbol, type, side, amount, price, {
>     'trailingPercent': 10,
>     'reduceOnly': True,
> })
> ```

That's not what I wanted, open with a stop loss I know how to do, close too, but the problem is that when I create the order with the stop loss, I don't immediately have the price of opening of my position, so my stoploss will be fake! That's why I want to create a position, after using a "fetch" to get my opening price and modify this same order with my correctly adjusted stop loss, I'm just looking for the code for " modify this same order with my correctly adjusted stop loss"

**Comment by Dan-krm** [comment]
@xerno0om what I explained isn't opening with an attached stop loss, it's adding a stop loss to a position after the position has already been opened.

1. use createOrder to open a position
2. use fetch to get your opening price
3. use createOrder with stopLossPrice to add a stop loss to your position

Using stopLossPrice on bingx is "modifying the same order with a correctly adjusted stop loss"
